pkg/config/mesh/mesh.go

```diff
@@ -117,6 +117,10 @@ func ApplyMeshConfig(yaml string, defaultConfig meshconfig.MeshConfig) (*meshcon
 	return &defaultConfig, nil
 }
 
+func MarshalMeshConfig(m *meshconfig.MeshConfig) (string, error) {
+	return gogoprotomarshal.ToYAML(m)
```

Comment:
Why is this roundtripping the yaml file? We just want to replicate the file.

Reply:
avoiding the roundtripping would involve major reworking of the mesh config file watching I think,since it just exposed the go struct not the original yaml file. Especially since the mesh config pilot is using may not even come from a file,there are a few different sources I believe. I don't think it's bad to round trip it, it's certainly not a performance issue at this scale? Are there other concerns?

Reply:
I don’t care about scale here. The original mesh config is hand written with comments. A stripped file is not an easy to modify. That said this is ok for now.

Reply:
you can't modify it anyways, we will overwrite it. you need to modify the source of truth in the "real" mesh config

Reply:
One of the points of namespace config, is that you *can* modify it by namespace.
However it we can achieve it in another way.

Reply:
I see. This is not that. However, we could build upon this. If you do the namespace stuff you need a fallback for namespaces without an override. So a user could mount there own config map and we would use that by default if present, otherwise use this?